test/scales/colorScaleTests.ts

```diff
@@ -0,0 +1,180 @@
+      ];
+
+      let scale: Plottable.Scales.Color;
+
+      beforeEach(() => {
+        scale = new Plottable.Scales.Color();
+      });
+
+      it("uses the default colors by default", () => {
+        assert.deepEqual(scale.range(), defaultColors, "The color scale uses the default colors by default");
+      });
+
+      it("uses altered colors if size of domain exceeds size of range", () => {
+        let colorRange = ["#5279c7", "#fd373e"];
+        scale.range(colorRange);
+        scale.domain(["a", "b", "c"]);
+        assert.strictEqual(scale.scale("a"), colorRange[0], "first color is used");
+        assert.strictEqual(scale.scale("b"), colorRange[1], "second color is used");
+
+        let alteredColor1Hex = scale.scale("c");
+        assert.notStrictEqual(alteredColor1Hex, colorRange[0], "first color has not been reused");
+        assert.notStrictEqual(alteredColor1Hex, colorRange[1], "second color has not been reused");
+        assert.notStrictEqual(alteredColor1Hex, "#000000", "the color does not fallback to black when running out of colors");
+        assert.notStrictEqual(alteredColor1Hex, "#ffffff", "the color does not fallback to white when running out of colors");
+
+        let color1 = TestMethods.colorHexToRGB(colorRange[0]);
+        let alteredColor1 = TestMethods.colorHexToRGB(alteredColor1Hex);
+        assert.operator(alteredColor1.red, ">", color1.red, "The resulting color should be lighter in the red component");
+        assert.operator(alteredColor1.green, ">", color1.green, "The resulting color should be lighter in the green component");
+        assert.operator(alteredColor1.blue, ">", color1.blue, "The resulting color should be lighter in the blue component");
+      });
+
+      it("interprets named color values correctly", () => {
+        scale.range(["red", "blue"]);
+        scale.domain(["a", "b"]);
+        assert.strictEqual(scale.scale("a"), "#ff0000", "red color as string should be correctly identified");
+        assert.strictEqual(scale.scale("b"), "#0000ff", "blue color as string should be correctly identified");
+        assert.deepEqual(scale.range(), ["red", "blue"], "the range itself does not convert to hex values");
+      });
+
+      it("accepts Category domain", () => {
+        scale.domain(["yes", "no", "maybe"]);
+        assert.strictEqual(scale.scale("yes"), defaultColors[0], "first color used for first option");
+        assert.strictEqual(scale.scale("no"), defaultColors[1], "second color used for second option");
+        assert.strictEqual(scale.scale("maybe"), defaultColors[2], "third color used for third option");
+      });
+    });
+
+    describe("CSS integration", () => {
+      let defaultColors = [
+        "#5279c7", "#fd373e", "#63c261", "#fad419", "#2c2b6f",
+        "#ff7939", "#db2e65", "#99ce50", "#962565", "#06cccc"
+      ];
+
+      it("accepts CSS specified colors", () => {
+        // Make sure you delete this at the end of the test
```

Comment:
it sounds a bit odd to me, maybe reasoning why it should be removed
talked to @bluong, and he suggested something along the line of "style should be removed before the test ends so it does not affect other tests"